src/Microsoft.DotNet.Wpf/src/PresentationCore/ThrowHelper.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+
+namespace PresentationCore;
+
+internal static class ThrowHelper
+{
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    [DoesNotReturn]
+    internal static void ThrowArgumentNullException(string paramName)
```

Comment:
Explicitly disallowing inlining for throwing, reduces the generated JIT massive. 
Inspired by dotnet/runtime